### PR TITLE
Include enum labels in YAML spec document and sort labels

### PR DESCRIPTION
**Pull request by kmadisa**
This is to extract the `enum_labels` property of the attributes and have them included in the generated yaml file. This will allow us to be able to validate the enum definitions of the attributes.

JIRA: [SAR-114](https://jira.skatelescope.org/browse/SAR-114)

### CODE REVIEW

**Comment by SKAJohanVenter** [comment]
@kmadisa  I also want to look at using `- name` as the top value (rather than alphabetical sort). Will comment soon.

File: tango_simlib/tango_yaml_tools/base.py [hunk 1]
```diff
@@ -42,43 +42,50 @@ def _build_yaml(self):
         for command in self.parser.get_device_command_metadata().values():
             command_data = {
                 "name": command["name"],
-                "dtype_in": command["dtype_in"].name,
-                "dtype_out": command["dtype_out"].name,
             }
-            for key in ["doc_out", "doc_in", "inherited", "disp_level"]:
+            # Order the rest alphabetically
+            for key in ["disp_level", "doc_in", "doc_out"]:
```
Comment (kmadisa) [inline tango_simlib/tango_yaml_tools/base.py:47]: Should we still keep these keys?

File: tango_simlib/tango_yaml_tools/base.py [hunk 2]
```diff
@@ -39,57 +39,51 @@ def _build_yaml(self):
...
+            attr_keys = sorted(attr.keys())
+            attr_keys.insert(0, attr_keys.pop(attr_keys.index("name")))
+            attr_data = {}
+            for key in attr_keys:
+                if key in ["data_format", "data_type", "disp_level"]:
+                    attr_data[key] = attr[key].name
+                elif key in [
+                        "name",
+                        "display_unit",
+                        "standard_unit",
```
Comment (ajoubertza) [inline tango_simlib/tango_yaml_tools/base.py:67]: I see `unit` has been removed, while we kept the other two.  I don't know which of the 3 are more commonly used, hopefully we aren't losing any useful info.

Comment (SKAJohanVenter) [reply tango_simlib/tango_yaml_tools/base.py:69]: I missed a few items, back to the original list

File: tango_simlib/tango_yaml_tools/base.py [hunk 3]
```diff
@@ -39,57 +39,61 @@ def _build_yaml(self):
...
+                        "max_warning",
+                        "description",
+                        "format",
+                        "delta_t",
+                        "max_alarm",
+                        "min_value",
+                        "inherited",
+                        "min_warning",
+                        "writable",
+                        "writable_attr_name",
```
Comment (kmadisa) [inline tango_simlib/tango_yaml_tools/base.py:85]: I think these keys should be discarded:

- writable_attr_name (result of having a writable attribute)
- inherited (POGO property)
- format  (by-product of the  data_type)
- label (used for GUIs)
- delta_val, delta_t, period (event related, which the values are not specified in the ICD)

Comment (SKAJohanVenter) [reply tango_simlib/tango_yaml_tools/base.py:85]: I'm conflicted, none of the keys are required. If you as a user want to add/drop any  it's up to you.

**Comment by kmadisa** [comment]
I'm happy to merge if that is all @SKAJohanVenter .

**Comment by SKAJohanVenter** [comment]
Don't merge yet

**Comment by SKAJohanVenter** [comment]
Sample file (had to use .txt):
[DishMaster.txt](https://github.com/ska-sa/tango-simlib/files/4938701/DishMaster.txt)

**Comment by kmadisa** [comment]
> Sample file (had to use .txt):
> [DishMaster.txt](https://github.com/ska-sa/tango-simlib/files/4938701/DishMaster.txt)
> @kmadisa

The names are no longer in alphabetical order :-(.

**Comment by SKAJohanVenter** [comment]
> > Sample file (had to use .txt):
> > [DishMaster.txt](https://github.com/ska-sa/tango-simlib/files/4938701/DishMaster.txt)
> > @kmadisa
> 
> The names are no longer in alphabetical order :-(.

It does uppercase first, then lowercase. Unless I'm missing something, it's late on a friday.

**Comment by kmadisa** [comment]
> > > Sample file (had to use .txt):
> > > [DishMaster.txt](https://github.com/ska-sa/tango-simlib/files/4938701/DishMaster.txt)
> > > @kmadisa
> > 
> > 
> > The names are no longer in alphabetical order :-(.
> 
> It does uppercase first, then lowercase. Unless I'm missing something, it's late on a friday.

Oh. I see. I wasn't looking at the upper/lower-case differences.

**Pull request merged**